public/app/features/alerting/unified/components/rules/deleted-rules/DeletedRules.tsx

```diff
@@ -0,0 +1,175 @@
+import { useState } from 'react';
+
+import { dateTimeFormat, dateTimeFormatTimeAgo } from '@grafana/data';
+import { Alert, Button, Column, InteractiveTable, LoadingPlaceholder, Stack } from '@grafana/ui';
+import { Trans, t } from 'app/core/internationalization';
+import { GrafanaRuleDefinition, RulerRuleDTO } from 'app/types/unified-alerting-dto';
+
+import { trackDeletedRuleRestoreFail, trackDeletedRuleRestoreSuccess } from '../../../Analytics';
+import { alertRuleApi } from '../../../api/alertRuleApi';
+import { GRAFANA_RULER_CONFIG } from '../../../api/featureDiscoveryApi';
+import { stringifyErrorLike } from '../../../utils/misc';
+import { rulerRuleType } from '../../../utils/rules';
+import { UpdatedByUser } from '../../rule-viewer/tabs/version-history/UpdatedBy';
+
+import { ConfirmRestoreDeletedRuleModal } from './ConfirmRestoreDeletedRuleModal';
+
+export const DELETED_RULES_PAGE_SIZE = 30;
+
+export function DeletedRules() {
```

Comment:
Can we add a few tests for this component?

Reply:
done!